File: TimeToToastApp/src/app/share/dtos/task/task-dto.ts
import { GuestDto } from "../guest/guest-dto";
import { UserDto } from "../user/user-dto";

export interface TaskDto {
    idTask?:number,
    description?:String,
    urgence?:number,
    dateTask?:Date,
    event?:Event,
    creator?:UserDto,
    assigneeUser?:UserDto,
    assigneeGuest?:GuestDto
}

File: TimeToToast_Spring/src/main/java/com/maddoxgraham/TimeToToast/DTOs/TaskDto.java
package com.maddoxgraham.TimeToToast.DTOs;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.LocalDate;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class TaskDto {
    private long description;
    private long urgence;
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private LocalDate dateTask;
    private long event;
    private long creator;
    private long assigneeUser;
    private long assigneeGuest;

}
